Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -147,18 +147,69 @@ Attaching to process 2670...
 
 #### Cycript and Cynject
 
-Cydia Substrate (formerly called MobileSubstrate) is the de-facto standard framework for developing run-time patches (“Cydia Substrate extensions”) on iOS. It comes with Cynject, a tool that provides code injection support for C. By injecting a JavaScriptCore VM into a running process on iOS, users can interface with C code, with support for primitive types, pointers, structs and C Strings, as well as Objective-C objects and data structures. It is also possible to access and instantiate Objective-C classes inside a running process. Some examples for the use of Cycript are listed in the iOS chapter.
+Cydia Substrate (formerly called MobileSubstrate) is the de-facto standard framework for developing run-time patches (“Cydia Substrate extensions”) on iOS. It comes with Cynject, a tool that provides code injection support for C. Cycript is a scripting language developed by Jay Freeman (saurik). Cycript injects a JavaScriptCore VM into the running process. Users can then manipulate the process using a hybrid of Objective-C++ and JavaScript syntax through the Cycript interactive console. It is also possible to access and instantiate Objective-C classes inside a running process. Some examples for the use of Cycript are listed in the iOS chapter.
 
-Cycript injects a JavaScriptCore VM into the running process. Users can then manipulate the process using JavaScript with some syntax extensions through the Cycript Console.
+Download SDK, unpack and Installation
 
--- TODO [Add use cases and example for Cycript] --
+	$ wget https://cydia.saurik.com/api/latest/3 -O cycript.zip && unzip cycript.zip
+	#on iphone
+	$ sudo cp -a Cycript.lib/*.dylib /usr/lib
+	$ sudo cp -a Cycript.lib/cycript-apl /usr/bin/cycript
 
-- Obtain references to existing objects
-- Instantiate objects from classes
-- Hooking native functions
-- Hooking objective-C methods
-- etc.*
-http://www.cycript.org/manual/
+To spawn the interactive cycript shell, you can run “./cyript” or just “cycript” if cycript is on your path.
+
+	$ cycyript
+	cy#
+
+To inject into a running process, we need to first find out the process ID (PID). We can run “cycript -p” with the PID to inject cycript into the process. To illustrate we will inject into springboard.
+
+	$ ps -ef | grep SpringBoard 
+	501 78 1 0 0:00.00 ?? 0:10.57 /System/Library/CoreServices/SpringBoard.app/SpringBoard
+	$ ./cycript -p 78
+	cy#
+We have injected cycript into SpringBoard, lets try to trigger an alert message on SpringBoard with cycript. 		
+
+	cy# alertView = [[UIAlertView alloc] initWithTitle:@"OWASP MSTG" message:@"Mobile Security Testing Guide"  delegate:nil cancelButtonitle:@"OK" otherButtonTitles:nil]
+	#"<UIAlertView: 0x1645c550; frame = (0 0; 0 0); layer = <CALayer: 0x164df160>>"
+	cy# [alertView show]
+	cy# [alertView release]
```

Comment:
Can you please format all code parts, like this:

```bash
# ls -alh
```

See also https://github.com/OWASP/owasp-mstg/blob/master/style_guide.md#code-and-shell-commands